.github/workflows/deploy-stacks.yml

```diff
@@ -0,0 +1,117 @@
+name: deploy-stacks.yml
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  create-env-change-sets-and-deploy:
+    permissions:
+      id-token: write
+      contents: read
+      issues: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node and install cdk cli
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Install dependencies and build project
+        run: |
+          npm install
+          npm install -g aws-cdk
+          npm run build
+
+      - name: Set up Env variables
+        run: |
+          echo "BETA_ACCOUNT_ID=${{ secrets.BETA_ACCOUNT_ID }}" >> $GITHUB_ENV
+          echo "PROD_ACCOUNT_ID=${{ secrets.PROD_ACCOUNT_ID }}" >> $GITHUB_ENV
+          echo "ASSUMED_ROLES=${{ secrets.ASSUMED_ROLES }}" >> $GITHUB_ENV
+          echo "REGION=${{ secrets.REGION }}" >> $GITHUB_ENV
+
+      - id: get_approvers
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: json
+          script: |
+            const collaborators = await github.paginate(
+              github.rest.repos.listCollaborators,
+              {
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                affiliation: 'all',
+                per_page: 100
+              }
+            );
+            const maintainers = collaborators.filter(c => c.permissions.maintain).map(c => c.login);
+            return maintainers;
```

Comment:
This would include everyone that has write access to the repository right? Or is is just maintain? 
Would it make more sense to parse the codeowners or maintainer file instead?

Reply:
This specifically filters users with maintainer access.

Reply:
Wasn't the case same here? https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/benchmark-pull-request.yml#L118-L126
All admins also added for approval https://github.com/opensearch-project/OpenSearch/issues/17904 right?

Reply:
fixed it by changing the `affiliation` parameter to `direct` instead of all. Can confirm it only lists maintainers for the repo.